app/Services/Model/Imp/SummitSponsorService.php

```diff
@@ -772,4 +774,60 @@ public function deleteSponsorSocialNetwork(Summit $summit, int $sponsor_id, int
             $summit_sponsor->removeSocialNetwork($social_network);
         });
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function addSponsorExtraQuestion(Summit $summit, int $sponsor_id, array $payload): SummitSponsorExtraQuestionType
+    {
+        return $this->tx_service->transaction(function () use ($summit, $sponsor_id, $payload) {
+            $summit_sponsor = $summit->getSummitSponsorById($sponsor_id);
+            if (is_null($summit_sponsor))
+                throw new EntityNotFoundException("Sponsor not found.");
+
+            $extra_question = SponsorExtraQuestionFactory::build($payload);
+
+            $summit_sponsor->addExtraQuestion($extra_question);
+
+            return $extra_question;
+        });
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function updateSponsorExtraQuestion(Summit $summit, int $sponsor_id, int $extra_question_id, array $payload): SummitSponsorExtraQuestionType
+    {
+        return $this->tx_service->transaction(function () use ($summit, $sponsor_id, $extra_question_id, $payload) {
+            $summit_sponsor = $summit->getSummitSponsorById($sponsor_id);
+            if (is_null($summit_sponsor))
+                throw new EntityNotFoundException("Sponsor not found.");
+
+            $extra_question = $summit_sponsor->getExtraQuestionById($extra_question_id);
```

Comment:
this is method is not updating question order
please check
https://github.com/OpenStackweb/summit-api/blob/a02088453cff6fb953998a34f15f66a9d107ce17/app/Services/Model/Imp/SummitOrderExtraQuestionTypeService.php#L170